lib/matplotlib/lines.py

```diff
@@ -1228,6 +1228,8 @@ def set_xdata(self, x):
         ----------
         x : 1D array
         """
+        if not np.iterable(x):
```

Comment:
- You can remove the corresponding checks in `__init__`.
- Do you want to check instead `np.ndim(x) == 1`?  (A 2D input doesn't really help either...)